moshi/src/test/java/com/squareup/moshi/KotlinExtensionsTest.kt

```diff
@@ -0,0 +1,64 @@
+package com.squareup.moshi
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import kotlin.annotation.AnnotationRetention.RUNTIME
+import kotlin.reflect.typeOf
+
+@JsonQualifier
+@Retention(RUNTIME)
+annotation class TestAnnotation1
+
+@JsonQualifier
+@Retention(RUNTIME)
+annotation class TestAnnotation2
+
+@TestAnnotation1
+@TestAnnotation2
+class KotlinExtensionsTest {
+
+  @Test
+  fun nextAnnotationsShouldWork() {
+    val annotations = KotlinExtensionsTest::class.java.annotations
+      .filterTo(mutableSetOf()) {
+        it.annotationClass.java.isAnnotationPresent(JsonQualifier::class.java)
+      }
+    assertEquals(2, annotations.size)
+    val next = annotations.nextAnnotations<TestAnnotation2>()
+    checkNotNull(next)
+    assertEquals(1, next.size)
+    assertTrue(next.first() is TestAnnotation1)
+  }
+
+  @Test
+  fun arrayType() {
+    val stringArray = String::class.asArrayType()
+    check(stringArray.genericComponentType == String::class.java)
+
+    val stringListType = typeOf<List<String>>()
+    val stringListArray = stringListType.asArrayType()
+    val expected = Types.arrayOf(Types.newParameterizedType(List::class.java, String::class.java))
+    assertEquals(stringListArray, expected)
+  }
+
+  @Test
+  fun addAdapterInferred() {
```

Comment:
Nice. Will it work if the type is parameterized?

Reply:
It should, I'll add a test to be sure

Reply:
Yup test worked ee1fe41